src/main/java/com/palantir/refreshable/DefaultRefreshable.java

```diff
@@ -178,42 +210,26 @@ private Disposable subscribeToSelf(Consumer<? super T> subscriber) {
      * // root should be garbage collected, nothing can update the SettableRefreshable value.
      * }</pre>
      */
-    private static final class DefaultDisposable implements Disposable {
-        private final Set<? extends Consumer<?>> subscribers;
-        private final Consumer<?> subscriber;
+    private static final class NotifyingDisposable<C extends Consumer<?>> implements Disposable {
+        private final DisposeListener<C> listener;
+        private final C subscriber;
 
-        DefaultDisposable(Set<? extends Consumer<?>> subscribers, Consumer<?> subscriber) {
-            this.subscribers = subscribers;
+        NotifyingDisposable(DisposeListener<C> listener, C subscriber) {
```

Comment:
Per comment on the other PR, I'd prefer to keep this as is and avoid additional lambda indirection. I think it makes the algorithm more mechanically obvious (harder to accidentally capture object references in future updates).

Reply:
Cool, in that case we defo want a small class for the other lambda.